packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cap.dart

```diff
@@ -6,27 +6,56 @@ import 'package:flutter/foundation.dart' show immutable;
 
 import 'types.dart';
 
+/// Enumeration of possible types of caps.
+enum CapType {
+  /// Cap that is squared off exactly at the start or end vertex of a [Polyline]
+  /// with solid stroke pattern, equivalent to having no additional cap beyond
+  /// the start or end vertex.
+  butt,
+
+  /// Cap that is a semicircle with radius equal to half the stroke width,
+  /// centered at the start or end vertex of a [Polyline] with solid stroke
+  /// pattern.
+  round,
+
+  /// Cap that is squared off after extending half the stroke width beyond the
+  /// start or end vertex of a [Polyline] with solid stroke pattern.
+  square,
+
+  /// CustomCap with a bitmap overlay centered at the start or
+  /// end vertex of a [Polyline], orientated according to the direction of the line's
+  /// first or last edge and scaled with respect to the line's stroke width.
+  custom,
+}
+
+String _capTypeToJson(CapType capType) => <CapType, String>{
+      CapType.butt: 'buttCap',
+      CapType.round: 'roundCap',
+      CapType.square: 'squareCap',
+      CapType.custom: 'customCap',
+    }[capType]!;
```

Comment:
Please use a switch instead; a switch would fail `analyze` if a new enum value were added, while this would not, making this version harder to maintain.

Reply:
I had not considered that, I will make the change and try to keep that in mind going forward. Thanks.